client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -719,6 +725,7 @@ class CalypsoifyIframe extends Component<
 
 		const isUsingClassicBlock = !! classicBlockEditorId;
 		const isCheckoutOverlayEnabled = config.isEnabled( 'post-editor/checkout-overlay' );
+		const { redirectTo, isFocusedLaunch, ...cartData } = checkoutModalOptions || {};
```

Comment:
Is there ever a case where we'd want to open the modal when `cartData` is empty and/or if it's missing `products`? If those should always be set here, maybe we should display an error or bail out early in some other way?

Reply:
Focused Launch is adding domain and plan to cart and then just opening Checkout Modal. See https://github.com/Automattic/wp-calypso/blob/trunk/packages/launch/src/hooks/use-cart.ts#L52-L75

If possible, would be good to switch that to using `cartData` as well in order to have a consistent API.

Reply:
(If it's not an easy question to answer, this is not high priority and could be addressed in a later change, I'm just curious.)

Reply:
Added https://github.com/Automattic/wp-calypso/issues/50463 to track this. While we explored this option, a domain product was not straight-forward to pass to `CompositeCheckout` from `EditorCheckoutModal` but I'm sure we'll figure it out with your help 🙇🏼